pkg/store/proxy.go

```diff
@@ -334,26 +346,43 @@ func (s *streamSeriesSet) Next() (ok bool) {
 		ctx = timeoutCtx
 	}
 
-	select {
-	case s.currSeries, ok = <-s.recvCh:
-		return ok
-	case <-ctx.Done():
+	if ctx.Err() != nil {
 		// closeSeries to shutdown a goroutine in startStreamSeriesSet.
 		s.closeSeries()
 
-		err := errors.Wrap(ctx.Err(), timeoutMsg)
+		wrapErr := errors.Wrap(ctx.Err(), timeoutMsg)
+		s.writeWarningOrErrorResponse(s.partialResponse, wrapErr)
+
 		if s.partialResponse {
-			level.Warn(s.logger).Log("err", err, "msg", "returning partial response")
-			s.warnCh.send(storepb.NewWarnSeriesResponse(err))
+			level.Warn(s.logger).Log("err", wrapErr, "msg", "returning partial response")
 			return false
 		}
-		s.errMtx.Lock()
-		s.err = err
-		s.errMtx.Unlock()
 
-		level.Warn(s.logger).Log("err", err, "msg", "partial response disabled; aborting request")
+		level.Warn(s.logger).Log("err", wrapErr, "msg", "partial response disabled; aborting request")
 		return false
 	}
+
+	s.currSeries, ok = <-s.recvCh
+
+	return ok
+}
+
+// writeWarningOrErrorResponse sends warning if partial response enabled or sets error otherwise
+func (s *streamSeriesSet) writeWarningOrErrorResponse(partialResponse bool, err error) {
```

Comment:
We can make this return nothing since we aren't using the return values anywhere and we aren't even setting them 😄